lib/OpenQA/Schema/Result/ScreenshotLinks.pm

```diff
@@ -36,8 +36,8 @@ __PACKAGE__->add_columns(
         is_nullable => 0,
     });
 
-__PACKAGE__->belongs_to(job        => 'OpenQA::Schema::Result::Jobs',        'job_id');
-__PACKAGE__->belongs_to(screenshot => 'OpenQA::Schema::Result::Screenshots', 'screenshot_id');
+__PACKAGE__->belongs_to(job        => 'OpenQA::Schema::Result::Jobs',        'job_id',        {on_delete => 'CASCADE'});
```

Comment:
I would believe this is too dangerous. If you happen to remove a job, you erase the screenshot links without the UPDATE in count.

Perhaps a postgresql trigger can help? If not, I'd prefer to drop the cascade

Reply:
You're right and this cascade delete (unlike the one in the next line) serves no purpose. So I'm dropping it. The cascade delete for screenshots is required but shouldn't be a problem because the screenshot is then gone anyways.

Reply:
One would sure hope so :)